File: tests/rolling_entropy_test.py
import numpy as np

from taflow import RollingEntropy


def test_rolling_entropy_lifecycle_and_alignment():
    state = RollingEntropy(timeperiod=2)
    state.extend([1.0, 2.0])
    np.testing.assert_allclose(state.compute(), [np.nan, np.log(2.0)], equal_nan=True)
    assert len(state) == 2
    state.reset()
    assert state.value is None

File: crates/taflow-core/src/stream/rolling_entropy_test.rs
use super::rolling_entropy::RollingEntropy;

#[test]
fn warmup_and_reset_are_consistent() {
    let mut state = RollingEntropy::new(2).unwrap();
    assert_eq!(state.append(1.0), None);
    assert!(state.append(2.0).is_some());
    state.reset();
    assert_eq!(state.value(), None);
}
